File: examples/identity/create_holder/create_holder.go
package main

import (
	"log"
	"os"

	"github.com/bloock/bloock-sdk-go/v2"
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/identity"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	// we set the API key and create a client
	bloock.ApiKey = os.Getenv("API_KEY")
	// we set de identity managed API host you have deployed
	bloock.IdentityApiHost = os.Getenv("IDENTITY_MANAGED_API_HOST")

	// initialize the Key Client
	keyClient := client.NewKeyClient()
	// initialize the IdentityClient
	identityClient := client.NewIdentityClient()

	// create the Baby JubJub local key
	localKey, err := keyClient.NewLocalKey(key.Bjj)
	if err != nil {
		log.Fatalln(err)
	}

	// we create the holder, passing the Baby JubJub key and holder did method
	holderKey := key.Key{LocalKey: &localKey}
	holderDidMethod := identity.PolygonID

	holder, err := identityClient.CreateHolder(holderKey, holderDidMethod)
	if err != nil {
		log.Fatalln(err)
	}

	log.Println(holder.Did.Did)       // will print the Holder DID public identifier. Ex: did:polygonid:polygon:main:2qCU58EJgrELSJT6EzT27Rw9DhvwamAdbMLpePztYq
	log.Println(holder.Did.DidMethod) // will return the DID method we chosed. Ex: identity.PolygonID
	log.Println(holder.Key)           // will return the Key BJJ object associated to the Holder.
}

File: examples/identity/create_holder/CreateHolder.java
import java.io.IOException;

import com.bloock.sdk.Bloock;
import com.bloock.sdk.client.IdentityClient;
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.entity.identity.DidMethod;
import com.bloock.sdk.entity.key.Key;
import com.bloock.sdk.entity.identity.Holder;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.LocalKey;

public class CreateHolder {
  public static void main(String[] args) throws Exception {
    try {
      // we set the API key and create a client
      Bloock.apiKey = System.getenv("API_KEY");
      // we set de identity managed API host you have deployed
      Bloock.identityApiHost = System.getenv("IDENTITY_MANAGED_API_HOST");

      // initialize the Key Client
      KeyClient keyClient = new KeyClient();
      // initialize the IdentityClient
      IdentityClient identityClient = new IdentityClient();

      // create the Baby JubJub local key
      LocalKey localKey = keyClient.newLocalKey(KeyType.Bjj);

      // we create the holder, passing the Baby JubJub key and holder did method
      Key holderKey = new Key(localKey);
      DidMethod holderDidMethod = DidMethod.PolygonID;

      Holder holder = identityClient.createHolder(holderKey, holderDidMethod);
      
      // will print the Holder DID public identifier. Ex:
      // did:polygonid:polygon:main:2qCU58EJgrELSJT6EzT27Rw9DhvwamAdbMLpePztYq
      System.out.println(holder.getDid().getDid());
      // will return the DID method we chosed. Ex: identity.PolygonID
      System.out.println(holder.getDid().getDidMethod());
      // will return the Key BJJ object associated to the Holder.
      System.out.println(holder.getKey());

    } catch (Exception e) {
      System.out.println(e);
    }
  }
}